### PR TITLE
complete rewrite

### PR DESCRIPTION
**Pull request by dgreisen-cfpb**

### CODE REVIEW

**Comment by dgreisen-cfpb** [comment]
I have reviewed @m3brown's work. @m3brown - please review and merge.

File: server/indexers/github_helpers.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ac_client = es_client.autocomplete
+history_client = es_client.history
+
+
+def save_indexed_version(gh_type, repo_name, typ, version):
+    doc_id = (gh_type + '/' + repo_name).replace('/', '%2F')
+    # TODO: create; update if error
+    resp = history_client._(typ)._(doc_id)._update.post(data={'version': version})
+    if resp.status_code == 500:
+        history_client._(typ)._(doc_id).put(data={'version': version})
```
Comment (m3brown) [inline server/indexers/github_helpers.py]: One concern I have about saving the version is that it's decoupled from the bulk_data save and I don't think it has to be.  Do you save the new version first, or the data? either way there's potential to get out of sync if there's an error in between.  Not a huge deal, but would it be better if save_indexed_version instead returned the json code to insert the data?

**Comment by m3brown** [comment]
Looks good, only had one minor comment that can be addressed later.

File: server/index.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+def index_gh_repos(repo_names=None, force=False):
+    repo_names = repo_names or _get_gh_repos()
+    jobs = [pool.spawn(indexers.wiki, 'GH', repo_name, gh_pool, force) for repo_name in repo_names]
+    jobs += [pool.spawn(indexers.readme, 'GH', repo_name, gh_pool, force) for repo_name in repo_names]
+    jobs += [pool.spawn(indexers.gh_pages, 'GH', repo_name, pages_pool, force) for repo_name in repo_names]
+    return jobs
+
+if __name__ == '__main__':
+    start = time.mktime(datetime.now().timetuple())
+    jobs = index_ghe_repos(['CFPB/handbook', 'SES/ses-next'], True)
```
Comment (m3brown) [inline server/index.py]: You might want to consider taking out the references to interal github repos.

**Pull request merged**